backend/agentd/session.go

```diff
@@ -207,58 +199,49 @@ func (s *Session) subPump() {
 				continue
 			}
 
-			msg := transport.NewMessage(corev2.CheckRequestType, configBytes)
-			s.sendq <- msg
-		case <-s.stopping:
+			msg = transport.NewMessage(corev2.CheckRequestType, configBytes)
+		case <-s.ctx.Done():
 			return
 		}
-	}
-}
-
-func (s *Session) sendPump() {
-	defer func() {
-		s.wg.Done()
-		logger.Info("shutting down - stopping sendPump")
-	}()
-
-	for {
-		select {
-		case msg := <-s.sendq:
-			logger.WithField("payload_size", len(msg.Payload)).Debug("session - sending message")
-			err := s.conn.Send(msg)
-			if err != nil {
-				switch err := err.(type) {
-				case transport.ConnectionError, transport.ClosedError:
-					return
-				default:
-					logger.WithError(err).Error("send error")
-				}
+		logger.WithField("payload_size", len(msg.Payload)).Debug("session - sending message")
+		if wsc, ok := s.conn.(*transport.WebSocketTransport); ok {
+			deadline := time.Now().Add(time.Second * time.Duration(s.cfg.WriteTimeout))
+			if err := wsc.Connection.SetWriteDeadline(deadline); err != nil {
```

Comment:
Are we calling `SetWriteDeadline()` before every message sent? If so, is that necessary?

Reply:
Yes, and I think it is. SetWriteDeadline takes a time.Time, so it needs to be constantly set to work properly AFAIK.

Reply:
I see, you're right!
See the comment above the `SetDeadline()` method of the [net.Conn interface](https://golang.org/pkg/net/#Conn).